prime-router/src/main/kotlin/azure/SendFunction.kt

```diff
@@ -155,9 +157,13 @@ class SendFunction(private val workflowEngine: WorkflowEngine = WorkflowEngine()
         actionHistory: ActionHistory,
         isEmptyBatch: Boolean,
     ): ReportEvent {
+        val reportContext = mapOf(
```

Comment:
Makes sense. I was trying to give fine-grained control of adding context params to specific log messages to be able to use `traces` to make graphs and visualizations. I've got custom events working well now so I think we can lean on that in a future PR for that use case.

I was concerned about the context leaking between requests but @mkalish assured me that by the nature of how an Azure functions app worked, that was basically impossible. I think if we want all logs to contain the same context we could just push directly to the MDC.

If we want certain blocks of logs to contain additional fields I can definitely add a similar function to `withLoggingContext`.

Is behavior like this what were looking for?
```kotlin

fun logging() {
    ThreadContext.put("everywhere", "value")
    logger.info("hello world!") // MDC = everywhere -> value
    withLoggingContext(mapOf("inside" to "other value")) {
        logger.info("inside") // MDC = everywhere -> value, inside -> other value
    }
    logger.info("back outside") // MDC = everywhere -> value
}
```

Reply:
Yes your example is exactly what I had in mind. I would just add that if another function got called inside of that one that it would also have the enriched logs.

Reply:
```
fun logging() {
    ThreadContext.put("everywhere", "value")
    logger.info("hello world!") // MDC = everywhere -> value
    withLoggingContext(mapOf("inside" to "other value")) {
        logger.info("inside") // MDC = everywhere -> value, inside -> other value
    }
    logger.info("back outside") // MDC = everywhere -> value
    logging_two()
}

fun logging_two() {
    logger.info("hello two") // MDC = everywhere -> value
}
```

Reply:
Sounds good to me!